### DISCUSSION TITLE
GET_LOCK query with alias

### DISCUSSION THREAD

**Discussion by thetutlage** [planetscale/discussion#307] [Bug Reports] [answered]
I have got a SQL query to obtain an advisory lock. Following is how the query looks like.

```sql
SELECT GET_LOCK('foo', 0) as lock_status;
```

In MySQL it returns the result where the column name is the alias `lock_status`. Check the following screenshot

<img width="720" alt="CleanShot 2022-10-15 at 12 16 00@2x" src="https://user-images.githubusercontent.com/1706381/195973409-8db147a8-9c69-474d-b85e-29277010bf06.png">

However, with Planetscale, the alias is not entertained. Again, the screenshot for reference.

<img width="873" alt="CleanShot 2022-10-15 at 12 15 52@2x" src="https://user-images.githubusercontent.com/1706381/195973431-8865b544-2e91-4a97-a76b-b3fefe19a20e.png">

Is it by design?

**Comment by orware** [comment, marked as answer]
Hello @thetutlage,

Thank you for reporting this to us!

Is the column alias a blocking issue within your project or is it able to be worked around using an alternative option for checking the return value in your application code?

The behavior definitely doesn't match what you are observing directly within MySQL and I was able to confirm the same incorrect behavior myself too so I have submitted an internal issue report with the details you've shared here, plus a few other details I collected, so this can be reviewed/corrected. The issue is likely occurring within the Vitess layer that sits on top of MySQL for PlanetScale databases. 

I hope this is helpful and that in the meanwhile you may be able to use a workaround due to the alias not being correctly respected in this case for your returned result set until a proper fix is available.

**Reply by thetutlage** [reply]
Nope, not a blocking issue. I just wanted to share the inconsistency
